integreat_cms/summ_ai_api/utils.py

```diff
@@ -0,0 +1,323 @@
+
+        :param text: The text to be translated
+        :type text: str
+        """
+        self.text = text
+
+    def translate(self, translated_text):
+        """
+        Translate the text of the current text field
+
+        :param translated_text: The translated text
+        :type translated_text: str
+        """
+        self.translated_text = translated_text
+
+    def __repr__(self):
+        """
+        The representation used for logging
+
+        :return: The canonical string representation of the text field
+        :rtype: str
+        """
+        return f"<{type(self).__name__} (text: {self.text})>"
+
+
+# pylint: disable=too-few-public-methods
+class HTMLSegment(TextField):
+    """
+    A class for translatable HTML segments
+    """
+
+    #: The current HTML segment
+    segment = None
+
+    # pylint: disable=super-init-not-called
+    def __init__(self, segment):
+        """
+        Convert the lxml tree element to a flat text string.
+        Preserve <br> tags as new lines characters.
+        Remove all inner tags but keep their text content.
+        Unescape all special HTML entities into unicode characters.
+
+        :param segment: The current HTML segment
+        :type segment: ~integreat_cms.cms.models.regions.region.Region
+        """
+        self.segment = segment
+        # Preserve new line tags
+        for br in self.segment.iter("br"):
+            br.tail = "\n" + br.tail if br.tail else "\n"
+        # Strip all inner tags
+        strip_tags(self.segment, "*")
+        # Unescape to convert umlauts etc. to unicode
+        self.text = unescape(self.segment.text_content())
+
+    def translate(self, translated_text):
+        """
+        Translate the current HTML segment and create new sub elements for line breaks
+
+        :param translated_text: The translated text
+        :type translated_text: str
+        """
+        # Split the text by newlines characters
+        lines = translated_text.splitlines()
+        # Take the first line as initial text
+        self.segment.text = lines[0]
+        # If there are more than one line returned, insert <br> tags
+        for line in lines[1:]:
+            SubElement(self.segment, "br").tail = line
+
+
+# pylint: disable=too-few-public-methods
+class HTMLField:
+    """
+    A class for more complex HTML fields which are splitted into segments
+    """
+
+    #: The list of HTML segments
+    segments = []
+    #: The current HTML stream
+    html = None
+
+    def __init__(self, html):
+        """
+        Parse the HTML string into an lxml tree object and split into segments
+
+        :param html: The HTML string content of this field
+        :type html: str
+        """
+        if html:
+            self.html = fromstring(html)
+            # Translate all <p> and <li> tags (and filter out empty segments)
+            self.segments = [
+                HTMLSegment(segment=segment) for segment in self.html.iter("p", "li")
```

Comment:
I was able to provoke an error, when I used quote and h2 elements (s. screenshot). I asked the service-team if these are commonly used elements or not. I haven't yet gotten an response. As soon as I have one, I'll comment here again :) 
![image](https://user-images.githubusercontent.com/72705147/191490669-78eb5f6f-4659-4112-af86-2f0a862358c6.png)

Reply:
I got a reply from the service-team, that h2 to h6 and quote-tags are rarely used. I think it's okay not to implement it now. If it comes up in the future it can still be implemented then